Docs/source/developers/testing.rst

```diff
@@ -47,97 +30,149 @@ You can run those locally before you push to save some time, install them once l
 
-There are three steps to follow to add a new automated test (illustrated here for PML boundary conditions):
+  .. code-block:: sh
 
-* An input file for your test, in folder `Example/Tests/...`. For the PML test, the input file is at ``Examples/Tests/pml/inputs_2d``. You can also re-use an existing input file (even better!) and pass specific parameters at runtime (see below).
-* A Python script that reads simulation output and tests correctness versus theory or calibrated results. For the PML test, see ``Examples/Tests/pml/analysis_pml_yee.py``. It typically ends with Python statement ``assert( error<0.01 )``.
-* If you need a new Python package dependency for testing, add it in ``Regression/requirements.txt``
-* Add an entry to ``Regression/WarpX-tests.ini``, so that a WarpX simulation runs your test in the continuous integration process, and the Python script is executed to assess the correctness. For the PML test, the entry is
+       add_warpx_test(
+           test_rz_laser_acceleration_picmi # name
+           RZ # dims
+           2  # nprocs
+           OFF # eb
+           "Examples/Tests/laser_acceleration/inputs_test_rz_laser_acceleration_picmi.py --test --dir 1" # inputs
+           Examples/Tests/laser_acceleration/analysis.py # analysis
+           diags/diag1/ # output (openPMD time series)
+           OFF # dependency
+       )
 
-.. code-block::
+If you need a new Python package dependency for testing, please add it in `Regression/requirements.txt <https://github.com/ECP-WarpX/WarpX/blob/development/Regression/requirements.txt>`__.
 
-   [pml_x_yee]
-   buildDir = .
-   inputFile = Examples/Tests/pml/inputs2d
-   runtime_params = warpx.do_dynamic_scheduling=0 algo.maxwell_solver=yee
-   dim = 2
-   addToCompileString =
-   cmakeSetupOpts = -DWarpX_DIMS=2
-   restartTest = 0
-   useMPI = 1
-   numprocs = 2
-   useOMP = 1
-   numthreads = 1
-   compileTest = 0
-   doVis = 0
-   analysisRoutine = Examples/Tests/pml/analysis_pml_yee.py
+Sometimes, two tests share a large number of input parameters. The shared input parameters can be collected in a "base" input file that can be passed as a runtime parameter in the actual test input files through the parameter ``FILE``.
 
-If you re-use an existing input file, you can add arguments to ``runtime_params``, like ``runtime_params = amr.max_level=1 amr.n_cell=32 512 max_step=100 plasma_e.zmin=-200.e-6``.
+Naming conventions for automated tests
+--------------------------------------
 
-.. note::
+Note that we currently obey the following snake\_case naming conventions for test names and test input files (which make automation tasks easier, e.g., parsing visually, parsing through code, sorting alphabetically, filtering tests in CTest via -R, etc.):
```

Comment:
```suggestion
Note that we currently obey the following snake\_case naming conventions for test names and test input files (which make automation tasks easier, e.g., parsing visually, parsing through code, sorting alphabetically, filtering tests in CTest via ``-R``, etc.):
```

Reply:
Done in [8e64870](https://github.com/ECP-WarpX/WarpX/pull/5068/commits/8e64870fe4fe42fd6c88f69741fe7761d30226c6).